File: modules/search/examples/autocomplete-sample.java
import com.couchbase.client.java.Cluster;
import com.couchbase.client.java.json.JsonObject;
import com.couchbase.client.java.search.SearchOptions;
import com.couchbase.client.java.search.SearchQuery;
import com.couchbase.client.java.search.result.SearchResult;
import com.couchbase.client.java.search.result.SearchRow;

public class MockTypeAheadEnGram {
  static String connstrn = System.getenv("$CB_HOSTNAME$"); 
  static String username = System.getenv("$CB_USERNAME$"); 
  static String password = System.getenv("$CB_PASSWORD$"); 
  
  public static void main(String... args) {
    Cluster cluster = Cluster.connect(connstrn, username, password);
    
    String iterStr = "Beaufort Hotel";
    for (int i=2; i<=Math.min(8, iterStr.length());i++) {
    	String testStr = iterStr.substring(0, i);
    	System.out.println("Input <" + testStr + ">, len: " + testStr.length());
    
    	SearchQuery query = SearchQuery.queryString(testStr);
    	SearchOptions options = SearchOptions.searchOptions().limit(8).skip(0).explain(false).fields("name");
    	SearchResult result = cluster.searchQuery("e-ngram-2-8" , query, options);
    	for (SearchRow row : result.rows()) {
    		System.out.println(row.id() + "," + row.fieldsAs(JsonObject.class) );
    	}
    }
  }
}

File: modules/search/examples/autocomplete-sample.js
const couchbase = require('couchbase')

const main = async () => {
  const hostname = process.env.CB_HOSTNAME
  const username = process.env.CB_USERNAME
  const password = process.env.CB_PASSWORD
  const bucketName = 'travel-sample'

  const cluster = await couchbase.connect(hostname, {username: username, password: password })

  const ftsMatchPrefix = async (term) => {
    try {
      const result =  await cluster.searchQuery(
        "e-ngram-2-8",
        couchbase.SearchQuery.queryString(term), { limit: 8, fields: ["name"] }
      )
      result.rows.forEach((row) => { console.log(row.id,row.fields)})
    } catch (error) {
      console.error(error)
    }
  }

  const inputStr = "Beaufort Hotel"
  for (let i = 2; i <= 8; i++) {
    var testStr = inputStr.substring(0, i);
    console.log("Input <" + testStr + ">, len: " + testStr.length)
    const res = await ftsMatchPrefix(testStr)
  }
}

main()

